cmake/FindLuaJIT.cmake

```diff
@@ -10,7 +10,7 @@
 find_path(LUA_INCLUDE_DIR luajit.h
   HINTS
     ENV LUA_DIR
-  PATH_SUFFIXES include/luajit-2.0 include/luajit-2.1 include
+  PATH_SUFFIXES include/luajit-2.0 include/luajit-2.1 include/luajit include
```

Comment:
also decided to finally take care of this. previously it wouldn't find luajit and we'd need to tell cmake where to find it explicitly. adjusted all documentation and CI stuff accordingly.